subprojects/docs/src/docs/userguide/jvm/toolchains.adoc

```diff
@@ -449,3 +449,14 @@ If an <<#sec:configuring_toolchain_specifications, unconfigured>> toolchain spec
 This can be achieved by passing an empty lambda when requesting a tool: `javaToolchainService.launcherFor({})`.
 
 You can find more details on defining custom tasks in the <<more_about_tasks.adoc#more_about_tasks, Authoring tasks>> documentation.
+
+[[sec:limitations]]
+== Toolchains limitations
+
+Gradle may detect toolchains incorrectly when a JVM/JRE compiled with `musl` library is used.
+The reason behind this is dynamic library loading mechanism.
+JVM/JRE may override `LD_LIBRARY_PATH` environment variable for proper library resolution.
+This might affect java subprocesses launched by Gradle, resulting in unexpected behavior.
+
+Thus, usage of multiple java toolchains is discouraged in environments with `musl` library.
```

Comment:
Is there a situation where toolchains will work when running in a `musl` JVM?  For instance, if I always specify toolchains with the same version as the JVM the build is running in, will that work?  It's not clear what distinction we are making with "multiple toolchains" here (i.e. does that mean it works if I have a single toolchain?)  We should either say "don't use toolchains with `musl`" or we should say "toolchains will work in a very specific scenario" and describe what that scenario is.  We should be clear about what we are suggesting (I think it's totally ok to say "just don't use toolchains" even though it might work in certain circumstances).

Reply:
First of all, thanks for review!

IIUC, Gradle would work fine if a single toolchain is used. I think this might be qualified as a pretty common scenario. In my experience, most companies don't care about multiple JVM versions support because their software would only be run in cloud/on-premise servers.